libnm-core/nm-setting-wireless.c

```diff
@@ -774,6 +755,27 @@ verify (NMSetting *setting, NMConnection *connection, GError **error)
 		return FALSE;
 	}
 
+	if (!priv->mode || strcmp (priv->mode, NM_SETTING_WIRELESS_MODE_INFRA) == 0) {
```

Comment:
why does it allow missing SSID for `NM_SETTING_WIRELESS_MODE_ADHOC`? That seems wrong and it should not check for INFRA mode, but instead explicitly check for not AP.

Reply:
I can't see why the Ad-Hoc networks shouldn't default to the hostname. On the contrary, it seems useful. At least that's what the Android handsets used to do iirc.